src/test-run/browser-manipulation-queue.js

```diff
@@ -1,75 +1,87 @@
-import { EventEmitter } from 'events';
 import shortId from 'shortid';
-import OS from 'os-family';
-import { resize as resizeWindow, getViewportSize } from 'testcafe-browser-natives';
+import { getViewportSize, generateThumbnail } from 'testcafe-browser-natives';
 import { isServiceCommand } from './commands/utils';
 import COMMAND_TYPE from './commands/type';
-import Warning from '../warnings';
+import WarningsLog from '../warnings/log';
 import WARNING_MESSAGE from '../warnings/message';
 
-export default class BrowserManipulationQueue extends EventEmitter {
-    constructor (windowId, screenshotCapturer) {
-        super();
 
+const SCREENSHOT_MANIPULATION_DESCRIPTION = 'take a screenshot';
+const RESIZE_MANIPULATION_DESCRIPTION     = 'do a window resize';
+
+export default class BrowserManipulationQueue {
+    constructor (browserConnection, screenshotCapturer) {
         this.commands           = [];
-        this.windowId           = windowId;
+        this.windowId           = browserConnection.id;
+        this.browserProvider    = browserConnection.provider;
         this.screenshotCapturer = screenshotCapturer;
+
+        this.warningLog = WarningsLog.get('manipulation:' + this.windowId);
```

Comment:
Just some examples:

1. `PathBrowserProvider` and `LocalBrowserProvider` do not support Linux yet, but others do. So I must implement some capability checking API in order to `BrowserManipulationQueue` be able to figure out which actions are actually supported by each provider.

2.  Currently `RemoteBrowserProvider` can perform a browser manipulation if the connected browser is actually a local browser. We discussed it with @AlexanderMoskovkin, and he considers it as a common scenario, when folks will bring their local browser as remote when using `testcafe` API with CI. Moreover, that is the way how our local functional tests are running. So it will be not nice, if I just tell "no, I don't support screenshot/resize", because I cannot be 100% sure,  can I do the requested action or not.